$OpenBSD: patch-mdb_c,v 1.6 2020/04/14 09:06:42 chrisz Exp $

enforce MDB_WRITEMAP, because we have no UBC (Unified Buffer Cache).

Index: mdb.c
--- mdb.c.orig
+++ mdb.c
@@ -4968,6 +4967,7 @@ mdb_env_open(MDB_env *env, const char *path, unsigned 
 		/* silently ignore WRITEMAP when we're only getting read access */
 		flags &= ~MDB_WRITEMAP;
 	} else {
+		flags |= MDB_WRITEMAP;
 		if (!((env->me_free_pgs = mdb_midl_alloc(MDB_IDL_UM_MAX)) &&
 			  (env->me_dirty_list = calloc(MDB_IDL_UM_SIZE, sizeof(MDB_ID2)))))
 			rc = ENOMEM;
